xarray/tests/test_dataset.py

```diff
@@ -1020,6 +1321,81 @@ def test_sel_points(self):
+                          coords={'a': ['a', 'b', 'c']})
+        idx_3 = DataArray(data['dim3'][[3, 2, 1]].values, dims=['a'],
+                          coords={'a': ['a', 'b', 'c']})
+        actual = data.sel(time=idx_t, dim2=idx_2, dim3=idx_3)
+        expected = data.isel(time=Variable(('a', ), [3, 2, 1]),
+                             dim2=Variable(('a', ), [3, 2, 1]),
+                             dim3=Variable(('a', ), [3, 2, 1]))
+        expected = expected.assign_coords(a=idx_t['a'])
+        self.assertDatasetIdentical(expected, actual)
+
+        idx_t = DataArray(data['time'][[3, 2, 1]].values, dims=['a'],
+                          coords={'a': ['a', 'b', 'c']})
+        idx_2 = DataArray(data['dim2'][[2, 1, 3]].values, dims=['b'],
+                          coords={'b': [0, 1, 2]})
+        idx_3 = DataArray(data['dim3'][[1, 2, 1]].values, dims=['c'],
+                          coords={'c': [0.0, 1.1, 2.2]})
+        actual = data.sel(time=idx_t, dim2=idx_2, dim3=idx_3)
+        expected = data.isel(time=Variable(('a', ), [3, 2, 1]),
+                             dim2=Variable(('b', ), [2, 1, 3]),
+                             dim3=Variable(('c', ), [1, 2, 1]))
+        expected = expected.assign_coords(a=idx_t['a'], b=idx_2['b'],
+                                          c=idx_3['c'])
+        self.assertDatasetIdentical(expected, actual)
+
+        # test from sel_points
+        data = Dataset({'foo': (('x', 'y'), np.arange(9).reshape(3, 3))})
+        data.coords.update({'x': [0, 1, 2], 'y': [0, 1, 2]})
+
+        expected = Dataset({'foo': ('points', [0, 4, 8])},
+                           coords={'x': Variable(('points', ), [0, 1, 2]),
+                                   'y': Variable(('points', ), [0, 1, 2])})
+        actual = data.sel(x=Variable(('points', ), [0, 1, 2]),
+                          y=Variable(('points', ), [0, 1, 2]))
+        self.assertDatasetIdentical(expected, actual)
+
+        expected.coords.update({'x': ('points', [0, 1, 2]),
+                                'y': ('points', [0, 1, 2])})
+        actual = data.sel(x=Variable(('points', ), [0.1, 1.1, 2.5]),
+                          y=Variable(('points', ), [0, 1.2, 2.0]),
+                          method='pad')
+        self.assertDatasetIdentical(expected, actual)
+
+        idx_x = DataArray([0, 1, 2], dims=['a'], coords={'a': ['a', 'b', 'c']})
+        idx_y = DataArray([0, 2, 1], dims=['b'], coords={'b': [0, 3, 6]})
+        expected_ary = data['foo'][[0, 1, 2], [0, 2, 1]]
+        actual = data.sel(x=idx_x, y=idx_y)
+        self.assertArrayEqual(expected_ary, actual['foo'])
+        self.assertDataArrayIdentical(actual['a'].drop('x'), idx_x['a'])
+        self.assertDataArrayIdentical(actual['b'].drop('y'), idx_y['b'])
+
+        if pd.__version__ >= '0.17':
+            with self.assertRaises(KeyError):
+                data.sel_points(x=[2.5], y=[2.0], method='pad', tolerance=1e-3)
+
```

Comment:
move to its own test and use `pytest.importorskip('pandas', minversion='0.17')` in place of your if statement